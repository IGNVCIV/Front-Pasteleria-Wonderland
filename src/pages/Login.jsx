// src/pages/Login.jsx
import { useState, useEffect } from "react";
import { useNavigate, useLocation, Link } from "react-router-dom";

function Login() {
  const [correo, setCorreo] = useState("");
  const [contrasena, setContrasena] = useState("");
  const [error, setError] = useState("");
  const navigate = useNavigate();
  const location = useLocation();
  const params = new URLSearchParams(location.search);
  const forceLogin = params.get("forceLogin"); // detecta si se fuerza login

  // Credenciales válidas (demo)
  const usuario = {
    correo: "admin@wonderland.cl",
    contrasena: "clave123",
  };

  useEffect(() => {
    const usuarioActivo = localStorage.getItem("usuarioActivo");
    const rol = localStorage.getItem("rol");

    // Redirige solo si hay sesión y NO se fuerza el login
    if (!forceLogin && usuarioActivo && rol === "admin") {
      navigate("/administracion");
    }
  }, [navigate, forceLogin]);

  const handleSubmit = (e) => {
    e.preventDefault();
    setError("");

    if (correo === usuario.correo && contrasena === usuario.contrasena) {
      localStorage.setItem("usuarioActivo", correo);
      localStorage.setItem("rol", "admin");
      navigate("/administracion");
    } else {
      setError("Usuario o contraseña incorrectos");
    }
  };

  return (
    <div className="login-page vh-100 d-flex flex-column">
      {/* HEADER */}
      <header>
        <nav className="nav">
          <Link to="/">
            <img
              src="/assets/img/Logos/Header.png"
              alt="Pastelería Wonderland"
              className="logo-img"
            />
          </Link>
        </nav>
      </header>

      {/* MAIN */}
      <main>
        <div className="wrap" id="Login">
          <div className="card shadow-sm">
            <div className="card-body">
              <h2>
                Inicio de Sesión<span className="title-underline"></span>
              </h2>

              {error && <div className="alert alert-danger">{error}</div>}

              <form onSubmit={handleSubmit} noValidate>
                <div className="field mb-3">
                  <label className="label" htmlFor="correo">
                    CORREO
                  </label>
                  <input
                    id="correo"
                    type="email"
                    className={`form-control ${error ? "is-invalid" : ""}`}
                    value={correo}
                    onChange={(e) => setCorreo(e.target.value)}
                  />
                </div>

                <div className="field mb-2">
                  <label className="label" htmlFor="contrasena">
                    CONTRASEÑA
                  </label>
                  <input
                    id="contrasena"
                    type="password"
                    className={`form-control ${error ? "is-invalid" : ""}`}
                    value={contrasena}
                    onChange={(e) => setContrasena(e.target.value)}
                  />
                </div>

                <div className="links mb-3 text-end">
                  <a href="#" className="small text-decoration-none">
                    ¿Olvidaste tu contraseña?
                  </a>
                </div>

                <div className="actions d-flex gap-2">
                  <button className="btn btn-primary flex-grow-1" type="submit">
                    Acceder
                  </button>
                  <button className="btn btn-outline flex-grow-1" type="button">
                    Crear cuenta
                  </button>
                </div>
              </form>
            </div>
          </div>
        </div>
      </main>

      {/* FOOTER */}
      <footer id="site-footer-login" className="mt-auto py-3">
        <div className="footer-content text-center">
          <p className="mb-1">&copy; Pastelería Wonderland — 2025</p>
          <a href="#" className="small text-decoration-none">
            Política de privacidad
          </a>
        </div>
      </footer>
    </div>
  );
}

export default Login;
