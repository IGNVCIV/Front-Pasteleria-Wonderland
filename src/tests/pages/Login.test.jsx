import '@testing-library/jest-dom';
import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, beforeEach } from "vitest";
import { MemoryRouter } from "react-router-dom";
import Login from "../../pages/Login";

// === TESTS ===
describe("Componente Login", () => {

  beforeEach(() => {
    localStorage.clear();
  });

  it("renderiza correctamente el título del formulario", () => {
    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );

    expect(screen.getByText(/Inicio de Sesión/i)).toBeInTheDocument();
  });

  it("muestra error al ingresar credenciales incorrectas", () => {
    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );

    const inputCorreo = screen.getByLabelText(/CORREO/i);
    const inputContrasena = screen.getByLabelText(/CONTRASEÑA/i);
    const botonAcceder = screen.getByRole("button", { name: /Acceder/i });

    fireEvent.change(inputCorreo, { target: { value: "mal@correo.com" } });
    fireEvent.change(inputContrasena, { target: { value: "incorrecta" } });
    fireEvent.click(botonAcceder);

    expect(screen.getByText("Usuario o contraseña incorrectos")).toBeInTheDocument();
  });

  it("guarda datos en localStorage al iniciar sesión correctamente", () => {
    render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );

    fireEvent.change(screen.getByLabelText(/CORREO/i), { target: { value: "admin@wonderland.cl" } });
    fireEvent.change(screen.getByLabelText(/CONTRASEÑA/i), { target: { value: "clave123" } });
    fireEvent.click(screen.getByRole("button", { name: /Acceder/i }));

    expect(localStorage.getItem("usuarioActivo")).toBe("admin@wonderland.cl");
    expect(localStorage.getItem("rol")).toBe("admin");
  });

  it("coincide con el snapshot actual", () => {
    const { container } = render(
      <MemoryRouter>
        <Login />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});


