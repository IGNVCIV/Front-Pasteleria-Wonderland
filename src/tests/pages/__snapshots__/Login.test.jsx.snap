// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Componente Login > coincide con el snapshot actual 1`] = `
<div>
  <div
    class="login-page vh-100 d-flex flex-column"
  >
    <header>
      <nav
        class="nav"
      >
        <a
          data-discover="true"
          href="/"
        >
          <img
            alt="Pastelería Wonderland"
            class="logo-img"
            src="/assets/img/Logos/Header.png"
          />
        </a>
      </nav>
    </header>
    <main>
      <div
        class="wrap"
        id="Login"
      >
        <div
          class="card shadow-sm"
        >
          <div
            class="card-body"
          >
            <h2>
              Inicio de Sesión
              <span
                class="title-underline"
              />
            </h2>
            <form
              novalidate=""
            >
              <div
                class="field mb-3"
              >
                <label
                  class="label"
                  for="correo"
                >
                  CORREO
                </label>
                <input
                  class="form-control "
                  id="correo"
                  type="email"
                  value=""
                />
              </div>
              <div
                class="field mb-2"
              >
                <label
                  class="label"
                  for="contrasena"
                >
                  CONTRASEÑA
                </label>
                <input
                  class="form-control "
                  id="contrasena"
                  type="password"
                  value=""
                />
              </div>
              <div
                class="links mb-3 text-end"
              >
                <a
                  class="small text-decoration-none"
                  href="#"
                >
                  ¿Olvidaste tu contraseña?
                </a>
              </div>
              <div
                class="actions d-flex gap-2"
              >
                <button
                  class="btn btn-primary flex-grow-1"
                  type="submit"
                >
                  Acceder
                </button>
                <button
                  class="btn btn-outline flex-grow-1"
                  type="button"
                >
                  Crear cuenta
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="mt-auto py-3"
      id="site-footer-login"
    >
      <div
        class="footer-content text-center"
      >
        <p
          class="mb-1"
        >
          © Pastelería Wonderland — 2025
        </p>
        <a
          class="small text-decoration-none"
          href="#"
        >
          Política de privacidad
        </a>
      </div>
    </footer>
  </div>
</div>
`;
